### ISSUE TITLE
AWS Provider - Feature Request: Elastic Beanstalk application version lifecycle.

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#537]
_This issue was originally opened by @beanaroo as hashicorp/terraform#11986. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

From AWS documentation on [Configuring Application Version Lifecycle Settings](http://docs.aws.amazon.com/elasticbeanstalk/latest/dg/applications-lifecycle.html)

The configuration parameter is [ApplicationResourceLifecycleConfig](http://docs.aws.amazon.com/elasticbeanstalk/latest/api/API_ApplicationResourceLifecycleConfig.html)

Maybe something along the lines of:
```
resource "aws_elastic_beanstalk_application" "tftest" {
  name = "tf-test-name"
  description = "tf-test-desc"
  appversion_lifecycle = {
    max_age = 7
    max_count = 350
    delete_from_s3 = true
  }
}
```

**Comment by robinbowes** [comment]
Would really like to see this added.

**Comment by Ray-B** [comment]
There is a PR for this now - is anyone available to review it?

**Comment by mrubin** [comment]
What is the status of this PR?

**Comment by krol3** [comment]
Also, I am interested in the status of this PR ! "appversion_lifecycle"

**Comment by bflad** [comment]
Sorry this one keeps slipping, that's my fault. I'll get it a proper review today or tomorrow.

**Comment by bflad** [comment]
Support for this configuration has been merged into master and will release with v1.16.0 of the AWS provider tomorrow.

**Comment by bflad** [comment]
This has been released in [version 1.16.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1160-april-25-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
